addons/calendar/models/calendar_alarm_manager.py

```diff
@@ -148,20 +157,26 @@ def _get_events_by_alarm_to_notify(self, alarm_type):
         already.
         """
         lastcall = self.env.context.get('lastcall', False) or fields.date.today() - relativedelta(weeks=1)
+        extra_conditions = self._get_notify_alert_extra_conditions()
         now = fields.Datetime.now()
-        self.env.cr.execute('''
-            SELECT "alarm"."id", "event"."id"
-              FROM "calendar_event" AS "event"
-              JOIN "calendar_alarm_calendar_event_rel" AS "event_alarm_rel"
-                ON "event"."id" = "event_alarm_rel"."calendar_event_id"
-              JOIN "calendar_alarm" AS "alarm"
-                ON "event_alarm_rel"."calendar_alarm_id" = "alarm"."id"
-             WHERE (
-                   "alarm"."alarm_type" = %s
-               AND "event"."active"
-               AND "event"."start" - CAST("alarm"."duration" || ' ' || "alarm"."interval" AS Interval) >= %s
-               AND "event"."start" - CAST("alarm"."duration" || ' ' || "alarm"."interval" AS Interval) < %s
-             )''', [alarm_type, lastcall, now])
+        self.env.cr.execute(SQL("""
+            SELECT alarm.id, event.id
+            FROM calendar_event AS event
+            JOIN calendar_alarm_calendar_event_rel AS event_alarm_rel
+                ON event.id = event_alarm_rel.calendar_event_id
+            JOIN calendar_alarm AS alarm
+                ON event_alarm_rel.calendar_alarm_id = alarm.id
+            WHERE alarm.alarm_type = %s
+            AND event.active
+            AND event.start - CAST(alarm.duration || ' ' || alarm.interval AS Interval) >= %s
+            AND event.start - CAST(alarm.duration || ' ' || alarm.interval AS Interval) < %s
```

Comment:
we could keep the initial format. 😇 

```suggestion
            SELECT alarm.id, event.id
              FROM calendar_event AS event
              JOIN calendar_alarm_calendar_event_rel AS event_alarm_rel
                ON event.id = event_alarm_rel.calendar_event_id
              JOIN calendar_alarm AS alarm
                ON event_alarm_rel.calendar_alarm_id = alarm.id
             WHERE alarm.alarm_type = %s
               AND event.active
               AND event.start - CAST(alarm.duration || ' ' || alarm.interval AS Interval) >= %s
               AND event.start - CAST(alarm.duration || ' ' || alarm.interval AS Interval) < %s
```